internal/journal/voljournal.go

```diff
@@ -281,13 +281,14 @@ func (conn *Connection) CheckReservation(ctx context.Context,
 	values, err := getOMapValues(
 		ctx, conn, journalPool, cj.namespace, cj.csiDirectory,
 		cj.commonPrefix, fetchKeys)
-	switch err.(type) {
-	case nil:
-	case util.ErrKeyNotFound, util.ErrPoolNotFound:
-		// pool or omap (oid) was not present
-		// stop processing but without an error for no reservation exists
-		return nil, nil
-	default:
+	if err != nil {
+		var eknf *util.ErrKeyNotFound
+		var epnf *util.ErrPoolNotFound
+		if errors.As(err, &eknf) || errors.As(err, &epnf) {
```

Comment:
A failure while creating an RBD volume:
```
omap.go:66] ID: 17 Req-ID: pvc-bd7ebc0d-3d40-4a63-8094-09e30d28466a omap not found (pool="replicapool", namespace="", name="csi.volumes.default"): rados: ret=2, No such file or directory
```
This suggests the return of `util.NewErrKeyNotFound()` from `getOMapValues()` does not work with `errors.As()`.

I must be missing something, https://github.com/golang/go/wiki/ErrorValueFAQ is pretty clear :thinking:

Reply:
found it :partying_face: The example in the [wiki](https://github.com/golang/go/wiki/ErrorValueFAQ#how-should-i-change-my-error-handling-code-to-work-with-the-new-features) shows a `var e *ErrSomeFailure`, but in most cases it should be `var e ErrSomeFailure` instead.